flair/models/sequence_tagger_model.py

```diff
@@ -587,11 +601,21 @@ def _viterbi_decode(self, feats):
             _, idx = torch.max(backscore, 0)
             prediction = idx.item()
             best_scores.append(softmax[prediction].item())
+            scores.append([elem.item() for elem in softmax.flatten()])
+        # This has been taken from https://github.com/zalandoresearch/flair/pull/642
```

Comment:
It is not entirely clear to me what this code actually does. I surrounded it with a simple assertion and it seems that scores remains unchanged. 

```py
 _scores_before = scores.copy()

swap_best_path, swap_max_score = (
    best_path[0],
    scores[-1].index(max(scores[-1])),
)

scores[-1][swap_best_path], scores[-1][swap_max_score] = (
    scores[-1][swap_max_score],
    scores[-1][swap_best_path],
)
assert _scores_before == scores
```

 @mauryaland Since you added this code in your original PR, can you perhaps shed some light on this? :)

Reply:
Sorry for the late reply! This is done when using CRF because sometimes the best path is not related to the max score.  However,I'm not sure that my implementation is optimal or always good.

Glad to see this feature added and thanks @Philipduerholt for having enhance and finish the few work I started.